server/src/main/java/org/opensearch/http/AbstractHttpServerTransport.java

```diff
@@ -352,17 +370,55 @@ protected void serverAcceptedChannel(HttpChannel httpChannel) {
      * @param httpChannel that received the http request
      */
     public void incomingRequest(final HttpRequest httpRequest, final HttpChannel httpChannel) {
-        handleIncomingRequest(httpRequest, httpChannel, httpRequest.getInboundException());
+        try (ThreadContext.StoredContext ignore = threadPool.getThreadContext().stashContext()) {
```

Comment:
What is the reason to stash context here?

Reply:
ThreadPool normally wraps the runnable but here threads are handled differently as daemon thread and the threadcontext/threadlocal is not being stashed post runnable complete. So if I am not adding this change then the span that we have started here doesn't get flushed post the request is complete anyways this is being ended as part or the TraceableHTTPChannel from some search thread.

Reply:
Understood, thank you, we may need to consider instrumenting OpenSearchExecutors::daemonThreadFactory, since it becomes the traceable asset as well

Reply:
Can I do this change in Separate PR?

Reply:
I am working on it. Will raise the PR in a while.

Reply:
Sure, I think the `stashContext` should work for now

Reply:
@Gaganjuneja since we redesigned the scopes, do we still need to stash the context here? I should work over thread locals, right?

Reply:
Ideally yes, let me test this one.

Reply:
It worked. Removed the stash context.